test/b-splines/linear.jl

```diff
@@ -61,4 +61,39 @@
         etp = extrapolate(itp, Flat())
         @test sum(isnan.(etp)) == 0
     end
+
+    @testset "Linear periodic" begin
+        # Constructors
+        @test Linear() === Linear(Throw(OnGrid()))
+        @test Linear() isa Linear{Throw{OnGrid}}
+        @test Linear(Periodic()) === Linear(Periodic(OnCell()))
+        @test Linear(Periodic()) isa Linear{Periodic{OnCell}}
+
```

Comment:
If we need `(deg::Linear)(gt::GridType)`, test it here.

```suggestion
        @test Linear(Periodic()) isa Linear{Periodic{OnCell}}
        @test Linear() === Linear()(OnGrid())

```

Reply:
I removed `(deg::Linear)(gt::GridType)`.